feat: Add Back to Top button to Contact page

tomive01888 opened the pull request:
---
# ✨ feat: Implement Back to Top button on Contact Page

### Overview

This PR introduces a basic "Back to Top" button on the `Contact` page. This is a preliminary implementation, and I've noted some key concerns and suggestions for its broader integration and robustness.

### Changes Made

* **Imported `BackToTopButton`:** Added 
```jsx
import ErrorBoundary from '../components/ErrorBoundary.jsx';
import BackToTopButton from '../components/buttons/BackToTop'; {/* <-- Import */}


    function Contact() {
      return (
        <ErrorBoundary>
          <div className="h-screen w-full flex justify-center items-center flex-col">
            <h1 className="text-2xl font-extrabold">Contact</h1>
            <p>Get in touch with us to learn more about how music changes lives.</p>
          </div>
          <BackToTopButton /> {/* <-- Button added here */}
        </ErrorBoundary>
      );
    }
```

### Key Concerns & Suggestions for Future Improvements

I've identified a few areas for enhancement that I'd recommend addressing:

* **Z-index Consistency:** The current `BackToTopButton` component lacks an explicit `z-index`. This may cause it to appear *underneath* other content (especially elements like the footer) on certain pages or screen sizes, impacting its visibility and usability.

* **Scroll Visibility Threshold:** In the `BackToTopButton` component's logic, the `setVisible` threshold is currently `window.scrollY > 300`. A minor usability improvement would be to change this to `window.scrollY >= 300`. This small adjustment would allow the button to appear immediately once the scroll reaches 300 pixels, rather than waiting for 301, making it feel slightly more responsive.

--- 

* **Global Component Placement:** The design file seems to mention having the "Back to Top" button across all pages. If that's the case, why not implement it higher up in the component tree (like in `MainLayout.jsx`)? This would let us reuse it more easily and cut down on redundant imports in multiple files. As a bonus, placing it in a global layout might also help resolve some `z-index` issues naturally, potentially reducing the need for explicit `z-index` changes on the button itself.

 **Suggested placement in `MainLayout.jsx`:** 
 ```jsx
    const MainLayout = () => {
      return (
        <>
          <div className="flex flex-col min-h-screen">
            <Navbar />
            <main className="flex-grow">
              <Outlet />
            </main>
            <Footer />
          </div>
          <BackToTopButton />  {/* <-- Button added here and resolves z-index issue too */}
        </>
      );
    };
    export default MainLayout;
```

tomive01888 committed:
feat: Add Back to Top button to Contact page

This commit introduces a basic "Back to Top" button specifically on the Contact page.

- Imported `BackToTopButton` component.
- Rendered `<BackToTopButton />` within the `Contact` component's DOM.

Concerns & Suggestions:
- Z-index: The button may not consistently stay on top of all content (e.g., footer) without explicit z-index.
- Global Placement: Recommend moving `BackToTopButton` to `MainLayout.jsx` for consistent, site-wide implementation as per design.
- Scroll Threshold: Suggest changing `window.scrollY > 300` to `>= 300` for quicker appearance on initial scroll.

senbet22 commented:
The animation feels a bit glitchy, and it sometimes clashes with the footer color, give me a call for better explanation :)

tomive01888 commented:
> The animation feels a bit glitchy, and it sometimes clashes with the footer color, give me a call for better explanation :)

I noticed the attempt to animate view of element using display:none and display:flex, which won't work. Guilty of trying something like this myself, so I know it doesn't work.  A fix would be to not use `style={{ display: visible ? 'flex' : 'none' }}` , cause you can't transition something that appears and disappears instantly. But directly use ternary operation and use the Tailwind CSS directly in className.

## Fix solution:

**A fix would be from:**
```jsx
<button
      type="button"
      onClick={scrollToTop}
      className="fixed bottom-8 right-8 bg-[var(--color-sunset-red)] w-[70px] h-[70px] p-4 rounded-full shadow-lg flex flex-col items-center justify-center transition-opacity duration-300 hover:opacity-90 cursor-pointer"
      style={{ display: visible ? 'opacity-100' : 'opacity-0' }}
      aria-label="Back to top"
    >
```

**to this:**
```jsx
<button
      type="button"
      onClick={scrollToTop}
      className={`fixed bottom-8 right-8 bg-[var(--color-sunset-red)] w-[70px] h-[70px] p-4 rounded-full shadow-lg flex flex-col items-center justify-center transition-opacity duration-300 hover:opacity-90 cursor-pointer ${visible ? 'opacity-100' : 'opacity-0'}`}
      aria-label="Back to top"
    >
``` 
This way we remove the line which aligns more to inline-style and not Tailwind styling. Instead, now we directly us opacity as ternary condition to render button's opacity on scroll with Tailwind CSS classes. For even more smoother animation rather than instantaneous appearing in the bottom of the corner.

--- 
### Help from the higher-ups:
Yes, I believe someone higher up needs to look into this and adjust color for this component and possibly allow component itself be placed higher preferably at MainLayout.jsx.

---

#### Possibly a future task for it?
I was not tasked to fix colors of it, maybe that is solved by next group or whoever wants to perhaps, personally for me I am not the best at colors. But it is something worth mentioning as the color choice of --color-sunset-red do clash with some of the other elements with same color.

tomive01888 committed:
Merge branch 'feature/contact-page-quantum-pixels' of https://github.com/NoroffFEU/musikkforandrerliv.no into feature/contact-back-to-top-button

tomive01888 committed:
refactor: Move Back to Top button to MainLayout for global access

Moves the `BackToTopButton` component from `Contact.jsx` to `MainLayout.jsx`.

This change ensures the button is present globally across all pages, aligning with design expectations and improving code reusability by eliminating the need for redundant imports on individual page components.

Also addresses potential z-index issues by placing the button at a higher level in the DOM tree.

---
** Issues that needs to be looked into **
- Animation effect not working
- Possibly adjust `setVisible(window.scrollY > 300)`to lower value or to be `>= 300` for quicker appearing on first scroll.
- Color issue with variable (--color-sunset-red) clashes with other elements of same color. Adjust darker shadow-color or simply make component a slightly darker tone, could use Tailwind class 'brightness' to adjust darkness.

senbet22 commented:
Good job Tom! keep up the good work
